sw/device/silicon_creator/rom/BUILD

```diff
@@ -159,65 +154,68 @@ cc_library(
     ],
 )
 
-opentitan_rom_binary(
+_ROM_DEVICES = [
+    "sim_dv",
+    "sim_verilator",
+    "fpga_cw310",
+    "fpga_cw340",
+]
+
+opentitan_binary(
     name = "rom_with_fake_keys",
     testonly = True,
+    exec_env = [
+        "//hw/top_earlgrey:fpga_cw310",
+        "//hw/top_earlgrey:fpga_cw340",
+        "//hw/top_earlgrey:sim_dv_base",
+        "//hw/top_earlgrey:sim_verilator_base",
+    ],
+    kind = "rom",
+    linker_script = ":linker_script",
     deps = [
         ":rom_without_keys",
         "//sw/device/silicon_creator/rom/keys/fake/rsa",
         "//sw/device/silicon_creator/rom/keys/fake/spx",
     ],
 )
 
-platform_target(
+# Create the legacy ROM target names so that existing opentitan_functest and
+# splicing rules can find the rom VMEM files.
+legacy_rom_targets(
+    testonly = True,
+    suffixes = _ROM_DEVICES,
+    target = "rom_with_fake_keys",
+)
+
+filegroup(
     name = "rom_with_fake_keys_fpga_cw310_elf",
     testonly = True,
-    platform = OPENTITAN_PLATFORM,
-    target = ":rom_with_fake_keys_fpga_cw310.elf",
+    srcs = [":rom_with_fake_keys"],
+    output_group = "fpga_cw310_elf",
 )
 
-opentitan_rom_binary(
-    name = "rom_with_real_keys_from_src",
+opentitan_binary(
+    name = "rom_with_real_keys",
+    exec_env = [
+        "//hw/top_earlgrey:fpga_cw310",
+        "//hw/top_earlgrey:fpga_cw340",
+        "//hw/top_earlgrey:sim_dv_base",
+        "//hw/top_earlgrey:sim_verilator_base",
+    ],
+    kind = "rom",
+    linker_script = ":linker_script",
     deps = [
         ":rom_without_keys",
         "//sw/device/silicon_creator/rom/keys/real/rsa",
         "//sw/device/silicon_creator/rom/keys/real/spx",
     ],
 )
 
-_ROM_DEVICES = [
-    "sim_dv",
-    "sim_verilator",
-    "fpga_cw310",
-]
-
-[
-    elf_to_scrambled_rom_vmem(
```

Comment:
Is this now done by default for each target using `opentitan_binary`? `sim_dv` probably needs it.

Reply:
Yes. For sim_dv it is done here: https://cs.opensource.google/opentitan/opentitan/+/master:rules/opentitan/sim_dv.bzl;l=45-55

Similar bits of code exist for sim_verilator and fpga_cw310, each producing the 39-bit scrambled vmem files.